graph_iip.go

```diff
@@ -40,10 +40,10 @@ func (n *Graph) sendIIPs() error {
 	// Send initial IPs
 	for i := range n.iips {
 		ip := n.iips[i]
+
 		// Get the receiver port channel
 		var channel reflect.Value
-		found := false
-		shouldClose := false
+		var found bool
```

Comment:
I'm wondering why `found := false` is a bad idea, given that it's a bit easier to read the explicit initial value, compared to the implicit default.

Reply:
You're right, `found := false` is equivalent to `var found bool`. It's more of a matter of personal preference, I like using default var values where possible. Whenever I see an assignment, I expect that to be for a purpose of initializing a var with a non-default value, so I kind of expect there to be a reason for such assignment. Absolutely not critical.
In this specific case, I first changed
```
found := false
shouldClose := false
```
to
```
var found, shouldClose bool
```
which reads a bit better than
```
found, shouldClose := false, false
```
Then I noticed that `shouldClose` is not used until later in the function and moved it further below to keep declaration close to the use, and that's how we ended up with this.

If you prefer, I can change initializations back to
```
found := false
```
and
```
shouldClose := false
```

Reply:
As I explained above, I prefer explicit value initialization over default zero, because this way the reader sees immediately the value the variable currently has, without having to infer what the default value of this type is.

Same goes to multi-assignment and multiple var declaration per line. E.g. ESLint community has a variety of rules to discourage it.

Reply:
Okay, no problem, let me revert this back.

Reply:
I agree with the in-place use of `shouldClose` though.

Reply:
Out of curiosity, I did a quick search to see if the golang community has a strong sentiment toward one or the other form of assignment and also the multi-declaration / multi-assignment. I could not find anything relevant in the [go code review comments](https://github.com/golang/go/wiki/CodeReviewComments), [effective go](https://golang.org/doc/effective_go.html), and I do not recall any linters complaining about the use of either form.

Karl Seguin in ["The little Go book"](https://github.com/karlseguin/the-little-go-book/blob/master/en/go.md#variables-and-declarations) writes:
>  ...you'll use `var NAME TYPE` when declaring a variable to its zero value, `NAME := VALUE` when declaring and assigning a value...

And Ivo Balbaert in "The way to Go":
> Other convenient shortening forms are:
> Multiple declarations of variables of the same type on a single line, like: `var a, b, c int`
> (this is an important reason why the type is written after the identifier(s) )
> Multiple assignments of variables on a single line, like: `a, b, c = 5, 7, “abc”`
> This assumes that variables a, b and c where already declared, if not: `a, b, c := 5, 7, “abc”`

So it looks like there is no strong preference in Go - unlike in JS / ESLint.

Reply:
Certainly, all of this makes sense. I just got curious :)